### PR TITLE
move node selector annotation to api repo

### PR DESCRIPTION
**Pull request by xichengliudui**
ref: https://github.com/istio/istio/pull/38167

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -300,6 +300,14 @@ annotations:
     resources:
       - Pod
 
+  - name: traffic.istio.io/nodeSelector
+    featureStatus: Stable
+    description:  the value for this annotation is a set of key value pairs (node labels) that can be used to select a subset of nodes from the pool of k8s nodes It is used for multi-cluster scenario, and with nodePort type gateway service.
+    deprecated: false
+    hidden: false
+    resources:
+      - Node
```
Comment (hzxuzhonghu) [inline annotation/annotations.yaml:309]: Not right, it is not applied to node

Comment (xichengliudui) [reply annotation/annotations.yaml:309]: Why is `nodeSelector` not a node? what should it be?

Comment (hzxuzhonghu) [reply annotation/annotations.yaml:309]: It is on a nodePort east-west gw svc.

Comment (xichengliudui) [reply annotation/annotations.yaml]: done

**Comment by xichengliudui** [comment]
cc @stevenctl

**Comment by xichengliudui** [comment]
cc @stevenctl

File: annotation/annotations.yaml [hunk 2]
```diff
@@ -300,6 +300,14 @@ annotations:
     resources:
       - Pod
 
+  - name: traffic.istio.io/nodeSelector
+    featureStatus: Stable
+    description:  the value for this annotation is a set of key value pairs (node labels) that can be used to select a subset of nodes from the pool of k8s nodes It is used for multi-cluster scenario, and with nodePort type gateway service.
```
Comment (hzxuzhonghu) [inline annotation/annotations.yaml:305]: The desc is not clear to me

Comment (stevenctl) [reply annotation/annotations.yaml:305]: Is this clearer: 

This annotation is a set of node-labels (ex: `key1=value,key2=value`). If the annotated Service is of type `NodePort` and is a multi-network gateway (see `topology.istio.io/network`), the addresses for selected nodes will be used for cross-network communication.

Comment (hzxuzhonghu) [reply annotation/annotations.yaml:305]: Looks good

**Review by stevenctl — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by xichengliudui** [comment]
@hzxuzhonghu 
It seems that a release-notes-none is needed.

**Comment by zirain** [comment]
> @hzxuzhonghu It seems that a release-notes-none is needed.

take it easy, reviewer will add `release-note-none` if it's needed.

**Comment by ericvn** [comment]
When a PR was opened, the creator can check a box that there are no user visible updates. If the box was checked at creation time, then the `release-notes-none` tag is automatically added. If the box isn't checked, it implies that there are user visible updates and that release notes are required so no label is added. Reviewers can update the label as well.

**Pull request merged**